Misc/NEWS.d/next/Library/2018-09-23-14-24-37.bpo-34776.1SrQe3.rst

```diff
@@ -0,0 +1 @@
+Fix dataclasses to support __future__ "annotations" mode
```

Comment:
Strictly speaking, this problem is not directly related to Postponed Evaluation of Annotations.

Consider rewording as "Fix dataclasses to support forward references in type annotations".

Here is an example that does not use PEP 563:

```python
from typing import get_type_hints
from dataclasses import dataclass

class T:
    pass

@dataclass()
class C2:
    x: 'T'

print(get_type_hints(C2.__init__))
```

Before your change:
```
Traceback (most recent call last):
  File ".\zzz.py", line 11, in <module>
    print(get_type_hints(C2.__init__))
  File "C:\Python37\lib\typing.py", line 1001, in get_type_hints
    value = _eval_type(value, globalns, localns)
  File "C:\Python37\lib\typing.py", line 260, in _eval_type
    return t._evaluate(globalns, localns)
  File "C:\Python37\lib\typing.py", line 464, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
NameError: name 'T' is not defined
```

After your change it works as expected:
```
{'x': <class '__main__.T'>, 'return': <class 'NoneType'>}
```